src/main/resources/ui/controller/UpsertWorkflow.controller.js

```diff
@@ -104,14 +104,27 @@ sap.ui.define([
 		},
 
 		onSaveWorkflow: function () {
-			let isEdit = this._model.getProperty("/isEdit");
-			let workflow = this.getWorkflowToSave();
-			if(isEdit) {
-				WorkflowRepository.updateWorkflow(workflow);
-			} else {
-				WorkflowRepository.createWorkflow(workflow);
+			if(this._isWorkflowValid(this._model.getProperty("/workflow"))){
+				let isEdit = this._model.getProperty("/isEdit");
+				let workflow = this.getWorkflowToSave();
+				if(isEdit) {
+					WorkflowRepository.updateWorkflow(workflow);
+				} else {
+					WorkflowRepository.createWorkflow(workflow);
+				}
+				this.myNavBack();
```

Comment:
The tabbing seems extensive. How about spaces and a max of 4?

Reply:
Will be fixed here: #60